### ISSUE TITLE
EntityFramework Handler does not work with EF5

### ISSUE THREAD

**Issue by arturomonge** [aws/aws-xray-sdk-dotnet#174]
The current EntityFramework Handler does not work with the new EntityFramework 5.0. **No exceptions are raised and no indications are given of any problem**. The handler simply does not work. 

After some research, the problem is generated because the [IDbCommandInterceptor](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.diagnostics.idbcommandinterceptor?view=efcore-5.0) interface has changed in EF Core 5.0. Most asynchronous methods now return a [ValueTask](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask-1?view=net-5.0).

**Comment by arturomonge** [comment]
I've opened the following pull requests to address this problem in two possible ways:

https://github.com/aws/aws-xray-sdk-dotnet/pull/175 : Upgrades the existing EntityFramework NuGet package with EF5 support and bumps version.
https://github.com/aws/aws-xray-sdk-dotnet/pull/176 : Adds a new EntityFramework5 NuGet package with EF5 support.

There is also a pull request that was previously opened here (https://github.com/aws/aws-xray-sdk-dotnet/pull/169), which attempts to solve the problem by targeting NET5.0. A few questions have been raised about the solution in this PR, and that is why I created two additional PRs to offer two additional options to solve this issue.

**Comment by lupengamzn** [comment]
Hey @arturomonge ,

Thank you for the contribution 

Currently we would like to keep the support for tracing Entity Framework in a single nuget package that customer can download to trace EF Core & EF 6. And implementing the [IDbCommandInterceptor](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.diagnostics.idbcommandinterceptor?view=efcore-5.0) for .NET 5 will introduce a lot of platform complexity with the current interceptor as you have mentioned that supported .NET platforms are kind of overlapping with each other. Supporting .NET 5 is on our roadmap and we hope to come up with a better solution to address this, e.g., switching to [DiagnosticListener](https://docs.microsoft.com/en-us/ef/core/logging-events-diagnostics/diagnostic-listeners). Sorry for the inconvenience and please stay tuned!

**Comment by arturomonge** [comment]
Understood, @lupengamzn. I'll close the pull requests and wait until you come up with a solution. Meanwhile, I created our own internal package specifically for EF5 using the modified sources, so we are not currently blocked by this.

**Comment by PauliusLe** [comment]
@lupengamzn what is the reason for closing this issue? I am also waiting for EF core 5 support

**Comment by arturomonge** [comment]
@lupengamzn I agree, this issue should not be closed, as this is still an outstanding request that needs to be addressed. I don't see any other issue open regarding EF Core 5 support.

**Comment by lupengamzn** [comment]
Hey @PauliusLe  @arturomonge ,

Sorry for the confusion and just reopened it for the purpose of tracking.

**Comment by btecu** [comment]
Any update on this?

**Comment by lupengamzn** [comment]
Hey @btecu ,

Currently there is no ETA on this, will leave this thread open for update.

**Comment by bruceharrison1984** [comment]
FYI, this is very easy to hack in by re-implementing a new DbInterceptor and copying a cut-down version of EFUtil in to your own project. Swapping Task<> to ValueTask<> is all it takes.

[DbInterceptor EF5 example](https://gist.github.com/bruceharrison1984/6cd2768fd2a3c865d17b16f29787b2fe)
[EFUtil EF5 example](https://gist.github.com/bruceharrison1984/8fff9633f3a7ea2c9f9b56361423e16e#file-hack-efutil-cs)

Add those to your project and register the Interceptor manually:
```csharp
services.AddDbContext<EcommContext>(options =>
{
    options.UseNpgsql(connectionString);
    options.AddInterceptors(new XRayInterceptor());
});
```

Works fine if you're not wanting to wait until support is officially added.
